### DISCUSSION TITLE
personal_sign request not resolved on mobile devices

### DISCUSSION THREAD

**Discussion by cgero-eth** [WalletConnect/walletconnect-monorepo#2384] [Developer Support]
When trying to sign a message using WalletConnect v2 on a mobile browser (Chrome or Safari), the request is successfully sent to the wallet (on TrustWallet for instance I get the "confirm-sign-message" dialog) but the dApp does not receive the signed message. The issue can be replicated on your [React test app](https://react-app.walletconnect.com/) by opening it on a mobile device, connecting the app to a wallet and clicking on the `personal_sign` button.

**Comment by finessevanes** [comment]
https://github.com/WalletConnect/walletconnect-monorepo/assets/44731696/fad81e50-7d1a-472d-b40f-d47d9e015685

Hey, tested the flow and you are correct, the dapp doesn't seem to get the message. I'm going to ping our team. Thanks for catching this!

**Comment by finessevanes** [comment]
Hey - Trust is aware of this issue, it's been fixed in the latest version 8.7 but it's currently blocked by Apple.

**Reply by AdemKao** [reply]
Hi @finessevanes,
I have to do some test on Mac and iPhone with your demo web(https://react-app.walletconnect.com), it sign message successfully on Mac with Trust Wallet.
But still has same issue on iPhone.

the version :
Mac : Apple M2 
OS: 13.3 
Trust Wallet :  8.7
Browser : Chrome
result: 
- [x] connect
- [x] sign
- [ ] sign action redirect to wallet app
<img width="329" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/53164941/92eb056d-890d-4537-97f3-8ab2a3d405c2">
<img width="393" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/53164941/ac2d2c90-4c59-48d8-94f7-b070cad822c1">
<hr/>

the version : 
Mac : Apple M2  
OS: 13.3  
Trust Wallet :  8.7 
Browser : Safari 
result:  
- [ ] connect
- [ ] sign
- [ ] sign action redirect to wallet app

could not choose Trust Wallet

<img width="372" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/53164941/82a6ca93-f7f9-4254-94fc-6892ebd1aac9">

<hr/>

the version : 
iPhone : SE  
OS: 16.1.2
Trust Wallet :  8.7 
Browser : Safari 
result:  
- [x] connect
- [ ] sign( Dapp could not get the result)
- [ ] sign action redirect to wallet app 

The Error : 
<img width="747" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/53164941/b0cb3c06-8bd9-4c94-9c45-8706914819e0">

<hr/>

I also  tested on other web(https://wagmi.sh/examples/sign-message)
it could redirect to Trust wallet after I click sign, but still can't get the sign result.

Currently I have no idea about how to interact with walletconnect and Trust wallet lol.

**Comment by cgero-eth** [comment]
@finessevanes Do you have an update on this? This is still not working on iOS version 8.8 released ~1week ago
